src/index.js

```diff
@@ -160,7 +161,7 @@ class UglifyJsPlugin {
                   'uglifyjs-webpack-plugin': versions.plugin,
                   'uglifyjs-webpack-plugin-options': this.options,
                   path: compiler.outputPath ? `${compiler.outputPath}/${file}` : file,
-                  input,
+                  inputHash: crypto.createHash('sha512').update(input).digest('hex'),
```

Comment:
`inputHash` => `hash`

Reply:
What other algos besides  `sha512` are available? (the shortest possible should be the way to go imho)

Reply:
Should we serialize the options separately and hash them aswell ?

Reply:
webpack appears to use [`md5`](https://github.com/webpack/webpack/search?utf8=%E2%9C%93&q=createHash+md5&type=) for other modules - probably for speed.

> Should we serialize the options separately and hash them aswell ?

Although the options could also be hashed, it may be easier to debug with a plain text prefix. A case could be made for either way.